### ISSUE TITLE
i18n.properties responds with wrong charset

### ISSUE THREAD

**Issue by htammen** [SAP/ui5-server#196]
## Expected Behavior
When using german umlaute like ä,ü,ö or other language specific chraracters in i18n.properties file it is expected that they are displayed as such in the running UI5 application.

## Current Behavior
When an Umlaut is used like in "**Anträge**" it is displayed as "**AntrÃ¤ge**".

## Steps to reproduce the issue
1. create a text resource in i18n.properties with an Umlaut, e.g. "requests=Anträge" and save the file with UTF-8 charset
2. use the text resource in a xml view of an UI5 app, e.g. &lt;Text text="{i18n>requests}" /&gt;
3. run the app

This happens when using "ui5 serve" to serve the development state of the app as well as when using "ui5 build" to build the app and then serve the content of the .dist folder with a webserver.

The i18n.properties file is delivered with `Content-Type: text/plain; charset=ISO-8859-1`. 
If I use e.g. WebIDE to run my application the i18n.properties has `Content-Type: text/plain`

When I save the i18n.properties with charset "ISO 8859-1" and use UI5 tooling to serve the app the result in the browser looks like this: **Antrï¿½ge**

## Context
- UI5 Module Version *(output of `ui5 --version` when using the CLI)*: `1.4.0`
- Node.js Version: `v10.15.0`
- npm Version: `6.6.0`
- OS/Platform: `Mac OS Mojave 10.14.4`
- Browser *(if relevant)*: `Chrome 74.0.3729.131, Firefox 66.0.3, Safari 12.1`
- Other information: 

## Affected components *(if known)*

- [ x ] [ui5-builder](https://github.com/SAP/ui5-builder)
- [ x ] [ui5-server](https://github.com/SAP/ui5-server)
- [ ] [ui5-cli](https://github.com/SAP/ui5-cli)
- [ ] [ui5-fs](https://github.com/SAP/ui5-fs)
- [ ] [ui5-project](https://github.com/SAP/ui5-project)
- [ ] [ui5-logger](https://github.com/SAP/ui5-logger)

**Comment by RandomByte** [comment]
Yes, you are right. This seems to be an issue with the usage of the replaceStream module here:

https://github.com/SAP/ui5-server/blob/d0e747d598b8f6696755581582f53e276260c72c/lib/middleware/serveResources.js#L62-L72

When removing the replaceStream-block, the client receives the correct data.

What bothers me is that I don't yet understand the behavior of replaceStream and Nodes [http.ServerResponse](https://nodejs.org/api/http.html#http_class_http_serverresponse) (which express' `res` object extends)


When supplying replaceStream with an `"encoding": "latin1"` option, it seems to convert the stream to UTF-8 (which kinda matches what [this headline promises](https://github.com/eugeneware/replacestream#changing-the-encoding)).  
The resulting stream consists of raw buffer chunks that return correct strings when logging them with `chunk.toString("utf8")` (=> `ü`) and corrupt strings with `chunk.toString("latin1")` (=> `Ã¼`).

Piping that stream into the express `res` object sends *something* (to be checked what is actually sent on the network) to the client. Note that the charset in the response header is still `ISO-8859-1`, which supposedly leads to the same corrupt string on the client (=> `Ã¼`). Changing the charset to `utf-8` in that case results in the correct string on the client (=> `ü`).

**Long story short**, we could exclude non UTF-8 encoded files from the version string replacement. We already exclude *.properties files when [*building* a UI5 project](https://github.com/SAP/ui5-builder/blob/master/lib/types/application/ApplicationBuilder.js#L44) anyways.

In addition, it turns out that [`mime-db`](https://github.com/jshttp/mime-db) (which we use through [`mime-types`](https://github.com/jshttp/mime-types)) [does not contain a single type that has a different charset than UTF-8](https://github.com/jshttp/mime-db/blob/master/db.json). So it's basically only *.properties files that are affected.

*Note that we use replaceStream also [in the UI5 Builder](https://github.com/SAP/ui5-builder/blob/389e6c88b8c38848e0b36a4008f6d11cb1cbf698/lib/processors/stringReplacer.js#L18).*

**Comment by RandomByte** [comment]
This should be resolved with UI5 CLI v1.5.3

**Comment by htammen** [comment]
Thx, works if I save the i18n.properties files with ISO-8859-1. Don't think that this is an international solution but it is ok for me in the European area
